python/src/aiconfig/scripts/aiconfig_cli.py

```diff
@@ -116,6 +139,24 @@ def _run_editor_servers(edit_config: EditServerConfig) -> Result[list[str], str]
 
 def _process_cli_config(cli_config: AIConfigCLIConfig) -> Result[bool, str]:
     LOGGER.setLevel(cli_config.log_level)
+
+    config_path = cli_config.config_path
+    file_contents = yaml.dump({"allow_usage_data_sharing": cli_config.allow_usage_data_sharing})
+    try:
+        with open(config_path, "x") as f:
+            f.write(file_contents)
+    except FileExistsError:
+        with open(config_path, "r") as f:
+            existing_file_contents = f.read()
+
+        existing_config = yaml.safe_load(existing_file_contents)
+        existing_config["allow_usage_data_sharing"] = cli_config.allow_usage_data_sharing
+        file_contents = yaml.dump(existing_config)
+        with open(config_path, "w") as f:
+            f.write(file_contents)
```

Comment:
Sorry what is this doing exactly in this function? Taking a step back out of thie catch block, we are:

1. Check if file exists --> if it does, use that param
2. If file does not exist --> write to that file

^Would this be cleaner to do? is there logic that I'm missing? Can we also have brief docstrings to explain this? Thanks

Reply:
#869 , going to resolve this